ludwig/models/llm.py

```diff
@@ -157,6 +157,10 @@ def __init__(
         set_pad_token(self.tokenizer)
 
         self.generation = GenerationConfig(**self.config_obj.generation.to_dict())
+        self.generation.pad_token_id = self.tokenizer.pad_token_id
```

Comment:
I believe that I have seen the situation -- with https://huggingface.co/mistralai/Mistral-7B-v0.1 if I am not mistaken -- that even the tokenizer of the base model has the `pad_token_id` as NULL; hence, it would be highly desirable if the proposed solution were verified against this recently published LLM (available on HuggingFace at the above URL).  Thank you.

Reply:
@alexsherstinsky So this will do the right thing actually!

The reason is that in the function called set_pad_token() above, we actually fallback to setting the pad token to the EOS token when the pad token is not set. We do this for Llama styled models as well. This is common practice since many models were pretrained without padding. 

So I do believe this should work? Were you running into issues when you tried it?

Reply:
@arnavgarg1 Thank you very much for this clarification.  I will try using this branch within 60 -- 120 minutes.  If by that time it gets merged, then I will try on the main branch.  I will keep you posted as to the outcome.  Thank you!